contribs/gmf/src/directives/timeslider.js

```diff
@@ -0,0 +1,284 @@
+   * @type (Array<number>|null)
+   */
+  this.timeValueList = this.getTimeValueList_();
+
+  /**
+   * Default Slider options (used by ui-slider directive)
+   * @type {{
+   *  range : boolean,
+   *  min : number,
+   *  max : number
+   * }}
+   * @export
+   */
+  this.sliderOptions = {
+    range : this.isModeRange,
+    min : this.minValue,
+    max : this.maxValue
+  };
+
+  /**
+   * Model for the ui-slider directive (date in ms format)
+   * @type {Array.<number>|number}
+   * @export
+   */
+  this.dates = this.isModeRange ? [initialOptions_.values[0], initialOptions_.values[1]] :
+  initialOptions_.values;
+
+};
+
+
+/**
+ * TimeSliderController.prototype.getTimeValueList_ - Get a list of time value instead
+ * of using the wmstime interval as a list of possibles values
+ * @private
+ * @return {Array<number>}  - List of timestamp representing possible values
+ */
+gmf.TimeSliderController.prototype.getTimeValueList_ = function() {
+  var wmsTime = this.time;
+  var timeValueList = null;
+  var minDate = new Date(this.minValue);
+  var maxDate = new Date(this.maxValue);
+
+  if (wmsTime.values) {
+    timeValueList = [];
+    wmsTime.values.forEach(function(date) {
+      timeValueList.push(new Date(date).getTime());
+    });
+  } else {
+    var maxNbValues = 1024;
+    var endDate = new Date(minDate.getTime());
+    endDate.setFullYear(minDate.getFullYear() + maxNbValues * wmsTime.interval[0]);
+    endDate.setMonth(minDate.getMonth() + maxNbValues * wmsTime.interval[1],
+      minDate.getDate() + maxNbValues * wmsTime.interval[2]);
+    endDate.setSeconds(minDate.getSeconds() + maxNbValues * wmsTime.interval[3]);
+
+    if (endDate > maxDate) {
+      // Transform interval to a list of values when the number
+      // of values is below a threshold (maxNbValues)
+      timeValueList = [];
+      for (var i = 0; ; i++) {
+        var nextDate = new Date(minDate.getTime());
+        nextDate.setFullYear(minDate.getFullYear() + i * wmsTime.interval[0]);
+        nextDate.setMonth(minDate.getMonth() + i * wmsTime.interval[1],
+          minDate.getDate() + i * wmsTime.interval[2]);
+        nextDate.setSeconds(minDate.getSeconds() + i * wmsTime.interval[3]);
+        if (nextDate <= maxDate) {
+          timeValueList.push(nextDate.getTime());
+        } else {
+          break;
+        }
+      }
+    }
+  }
+  return timeValueList;
+};
+
+
+/**
+ * Compute the closest available date from the given timestamp
+ * @param  {number} timestamp selected datetime (in ms format)
+ * @return {number} the closest available datetime (in ms format) from the timestamp
+ * @private
+ */
+gmf.TimeSliderController.prototype.getClosestValue_ = function(timestamp) {
+  if (timestamp <= this.minValue) {
+    return this.minValue;
+  }
+
+  if (timestamp >= this.maxValue) {
+    return this.maxValue;
+  }
+
+  if (this.timeValueList) {
+    // Time stops are defined as a list of values
+    var idx;
```

Comment:
IMO this is really not important. Indeed even if I name them rightDistance rightIndex... you'll not be able to understand the code easily without taking time to fully understand the behavior. And if you take time to understand the code, rIdx, lIdx rDist... is obvious. But if it's important for you, I can rename those variables.